drone_control_gym.py

```diff
@@ -126,14 +126,21 @@ def _calculate_reward(self):
             return True, GOAL_REWARD
         else:
             reward = 0
+            finished = False
             # if drone is flipped (or collided?), return FLIPPED_REWARD
             if abs(self.drone_rpy[0]) > ROLL_THRESHOLD or abs(self.drone_rpy[1]) > PITCH_THRESHOLD:
                 logging.debug("drone has flipped")
                 reward += FLIPPED_REWARD
+                finished = True
             # if drone is out of RANGE_LIMIT return OUT_OF_BOUND_REWARD, not sure need this or not
-            if self.drone_pose[0] > RANGE_LIMIT or self.drone_pose[1] > RANGE_LIMIT:
+            if (
+                self.drone_pose[0] > RANGE_LIMIT
+                or self.drone_pose[1] > RANGE_LIMIT
+                or self.drone_pose[2] < HEIGHT_LOWER_LIMIT
```

Comment:
Drone will start from ground, this check will always fullfil? i think we can remove this. The drone base_link is the drone main body's bottom surface. So z will be 0 when it is spawn on the ground plane.